kornia/augmentation/augmentation.py

```diff
@@ -1369,3 +1370,46 @@ def generate_parameters(self, batch_shape: torch.Size) -> Dict[str, torch.Tensor
 
     def apply_transform(self, input: torch.Tensor, params: Dict[str, torch.Tensor]) -> torch.Tensor:
         return self.transform(input)
+
+
+class RandomInvert(AugmentationBase2D):
+
+    r"""Invert the tensor images values randomly.
+
+    Args:
+        max_val (torch.Tensor): The expected maximum value in the input tensor. The shape has to
+          according to the input tensor shape, or at least has to work with broadcasting. Default: 1.0.
+        return_transform (bool): if ``True`` return the matrix describing the transformation applied to each
+            input tensor. If ``False`` and the input is a tuple the applied transformation wont be concatenated.
+        same_on_batch (bool): apply the same transformation across the batch. Default: False.
+        p (float): probability of applying the transformation. Default value is 0.5.
+
+    Examples:
+        >>> rng = torch.manual_seed(0)
+        >>> img = torch.rand(1, 1, 5, 5)
+        >>> inv = RandomInvert()
+        >>> inv(img)
+        tensor([[[[0.4963, 0.7682, 0.0885, 0.1320, 0.3074],
+                  [0.6341, 0.4901, 0.8964, 0.4556, 0.6323],
+                  [0.3489, 0.4017, 0.0223, 0.1689, 0.2939],
+                  [0.5185, 0.6977, 0.8000, 0.1610, 0.2823],
+                  [0.6816, 0.9152, 0.3971, 0.8742, 0.4194]]]])
+    """
+
+    def __init__(self,
+                 max_val: torch.Tensor = torch.tensor(1.),
```

Comment:
Shouldn't be a Union type? ```int, float, tensor```?

Reply:
if any users asks we can implement later